### PR TITLE
Issue/5801 simple payments link

### LINKED ISSUE #5801: [Simple Payments]: Add payment link
The iOS app adds a payment link to the simple payments take payment screen and WCAndroid should do the same. Note that this feature must be marked as BETA - explanation [in this iOS PR](https://github.com/woocommerce/woocommerce-ios/pull/6064)

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #5801 - adds a "Share payment link" row to the simple payment flow to enable users to easily share the link to taking payment. Note that this requires WooCommerce 6.4 for testing since the `payment_url` field wasn't available until that version.

This means either using the WooCommerce beta tester plugin to install the latest 6.4 RC, or downloading the RC [from here](https://developer.woocommerce.com/2022/03/29/woocommerce-6-4-rc/) and uploading it manually.

To test:

* Before updating to the 6.4 RC, step through the simple payment flow and verify the "Share payment link" does _not_ appear
* Update to the 6.4 RC and verify that it does show, and tapping it enables sharing the link

https://user-images.githubusercontent.com/3903757/163035904-7c144470-d8ed-4252-9146-2dde02928275.mp4

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/simplepayments/TakePaymentViewModel.kt [hunk 1]
```diff
@@ -34,12 +37,21 @@ class TakePaymentViewModel @Inject constructor(
 ) : ScopedViewModel(savedState) {
     private val navArgs: TakePaymentFragmentArgs by savedState.navArgs()
 
+    val viewStateLiveData = LiveDataDelegate(savedState, ViewState())
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/simplepayments/TakePaymentViewModel.kt]: 🔧 WDYT about not introducing observable ViewState and just take needed value from `val order`?

`order` is not a subject to change on this screen - when selected, we don't have to observe for any changes. Using `order#paymentUrl` directly in fragment's `onCreateView` will simplify this ViewModel.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/simplepayments/TakePaymentViewModel.kt [hunk 2]
```diff
@@ -34,12 +37,21 @@ class TakePaymentViewModel @Inject constructor(
...
+    internal var viewState by viewStateLiveData
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/simplepayments/TakePaymentViewModel.kt]: 🔧 `internal` is not very useful here, probably private is the fit
```suggestion
    private var viewState by viewStateLiveData
```

**Review by wzieba — COMMENTED** [review]
👋 Works good but I think we can simplify things by removing `ViewState`, WDYT?

And I've noticed that navigating back from the share screen puts the user to the list of orders - maybe we should navigate back to "take payment options" instead?

https://user-images.githubusercontent.com/5845095/163364558-6e508dc6-0e28-4610-9357-68c837743ca8.mov

**Comment by nbradbury** [comment]
@wzieba Thanks for the review! I've made the suggested changes, except for navigating back to the "take payment" screen when the user backs out of the share sheet. 

I initially had it only leave that fragment when `registerForActivityResult` returned `RESULT_OK`, but that turned out to be unreliable. It returns `RESULT_OK` when the user chooses to copy the URL to the clipboard, but sometimes it resturns `RESULT_CANCELED` even if the user successfully shared the URL (Gmail, for example, does this).

So the choice was to either always return to the order list, or always return to "Take payment," and I opted for the former.

**Comment by wzieba** [comment]
I see, it works really strange also with e.g. Facebook Messenger 🤷 . After sharing I had to navigate back and the result was `RESULT_CANCELED`.

Looks good, CI is failing but only CircleCI and it's not related to this PR (p1649933536095639-slack-CC7L49W13).

:shipit:

**Review by wzieba — APPROVED** [review]
Thanks for adding improvements!

**Comment by wzieba** [comment]
Oh I see that CircleCI is `required` step. So I think we have to wait for working CircleCI.

**Pull request merged**